tests/unit/s2n_server_renegotiation_info_test.c

```diff
@@ -30,49 +30,99 @@ int main(int argc, char **argv)
 {
     BEGIN_TEST();
 
-    struct s2n_config *config;
-    EXPECT_NOT_NULL(config = s2n_config_new());
+    /* Test should_send */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+
+        /* TLS1.2 and secure renegotiation not enabled -> DON'T send */
+        conn->actual_protocol_version = S2N_TLS12;
+        conn->secure_renegotiation = false;
+        EXPECT_FALSE(s2n_server_renegotiation_info_extension.should_send(conn));
+
+        /* TLS1.3 and secure renegotiation not enabled -> DON'T send */
+        conn->actual_protocol_version = S2N_TLS13;
+        conn->secure_renegotiation = false;
+        EXPECT_FALSE(s2n_server_renegotiation_info_extension.should_send(conn));
+
+        /* TLS1.3 and secure renegotiation enabled -> DON'T send */
+        conn->actual_protocol_version = S2N_TLS13;
+        conn->secure_renegotiation = true;
+        EXPECT_FALSE(s2n_server_renegotiation_info_extension.should_send(conn));
+
+        /* TLS1.2 and secure renegotiation enabled -> send! */
```

Comment:
Another send! :D 

Also, shouldnt we be sending  when version is TLS1.2 and secure renegotiation enabled?

Reply:
Sending is just so exciting!